Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h

```diff
@@ -286,6 +290,7 @@ class NetworkStorageManager final : public IPC::WorkQueueMessageReceiver, public
     UnifiedOriginStorageLevel m_unifiedOriginStorageLevel;
     IPC::Connection::UniqueID m_parentConnection;
     HashMap<IPC::Connection::UniqueID, HashSet<String>> m_temporaryBlobPathsByConnection WTF_GUARDED_BY_CAPABILITY(workQueue());
+    HashMap<IPC::Connection::UniqueID, HashSet<String>> m_allowedFilePathsByConnection WTF_GUARDED_BY_CAPABILITY(workQueue());
```

Comment:
`m_allowedFilePathsByConnection` should be accessed on main thread, not work queue because there is only one case where the background queue operation will change it: `NetworkStorageManager::getFile()`; all the other cases (`NetworkConnectionToWebProcess::registerInternalFileBlobURL()`, `NetworkProcess::allowFileAccessFromWebProcess()`, etc) are on main thread.

Also, I don't think we need to add the path when File System handle is created, since only `FileSystemFileHandle::getFile()` will create file backed blob. That means we only need to make `NetworkStorageManager::getFile()` dispatch a task to main thread to add the path to `m_allowedFilePathsByConnection`.